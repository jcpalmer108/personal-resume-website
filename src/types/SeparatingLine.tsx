export type SeparatingLineProps = {
  thin?: boolean;
  light?: boolean;
  always?: boolean;
};
