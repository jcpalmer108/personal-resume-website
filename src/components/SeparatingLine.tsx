import { Line } from "../styles/components/SeparatingLine";
import { SeparatingLineProps } from "../types/SeparatingLine";

export default function SeparatingLine({
  light,
  thin,
  always,
}: SeparatingLineProps) {
  return <Line light={light} thin={thin} always={always} data-testid="Line" />;
}
