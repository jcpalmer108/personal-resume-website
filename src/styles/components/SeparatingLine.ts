import styled from "styled-components";
import { desktopBreakpoint, silver_polish } from "../variables";
import { SeparatingLineProps as LineProps } from "../../types/SeparatingLine";

export const Line = styled.hr<LineProps>`
  width: calc(100% - 2px);
  border-color: ${(props) => {
    if (props.always && props.light) {
      return "white";
    } else if (props.always) {
      return silver_polish;
    } else {
      return "transparent";
    }
  }};
  border-width: ${(props) => (props.thin ? 0.5 : 1)}px;
  border-block-start: none;
  margin: 0px;

  @media (min-width: ${desktopBreakpoint}px) {
    border-color: ${(props) => (props.light ? "white" : silver_polish)};
    border-width: ${(props) => (props.thin ? 0.5 : 1)}px;
    position: absolute;
    left: 0px;
    margin: 0px;
  }
`;
