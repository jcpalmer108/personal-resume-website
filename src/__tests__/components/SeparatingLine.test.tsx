import { render, screen } from "@testing-library/react";
import SeparatingLine from "../../components/SeparatingLine";

// params for component
const params = {
  optional: {
    light: true,
    thin: true,
    always: true,
  },
};
describe("SeparatingLine", () => {
  test("renders if optional params are passed in", () => {
    // given
    const { optional } = params;
    render(
      <SeparatingLine
        light={optional.light}
        thin={optional.thin}
        always={optional.always}
      />
    );

    // then
    expect(screen.getByTestId("Line")).toMatchSnapshot();
  });

  test("renders if optional params are not passed in", () => {
    // given
    render(<SeparatingLine />);

    // then
    expect(screen.getByTestId("Line")).toMatchSnapshot();
  });
});
