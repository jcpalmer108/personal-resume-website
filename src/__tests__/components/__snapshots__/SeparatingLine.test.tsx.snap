// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeparatingLine renders if optional params are not passed in 1`] = `
<hr
  class="sc-bczRLJ iPbZrq"
  data-testid="Line"
/>
`;

exports[`SeparatingLine renders if optional params are passed in 1`] = `
<hr
  class="sc-bczRLJ bLYoYx"
  data-testid="Line"
/>
`;
